libraries/mbed/common/rtc_time.c

```diff
@@ -29,15 +41,18 @@ time_t time(time_t *timer)
 #endif
 
 {
-#if DEVICE_RTC
-    if (!(rtc_isenabled())) {
-        set_time(0);
+    if (_rtc_isenabled != NULL) {
+        if (!(_rtc_isenabled())) {
+            set_time(0);
+        }
+    }
+    
+    time_t t;
+    if (_rtc_read != NULL) {
+        t = _rtc_read();
+    } else {
+        t = 0;
```

Comment:
I guess bit cleaner code, and tiny, tiny, bit slower. If that is preferred I will change it.